doc/release/README.executing.rst

```diff
@@ -0,0 +1,369 @@
+Chapel program, as well as a list of the configuration variables
+defined by the program and their types.  If the configuration variable
+has been set on the command line, its value is also shown.
+
+
+-------------------------------
+Setting Configuration Variables
+-------------------------------
+
+Configuration constants and variables defined in a Chapel program can
+have their default values overridden on the command line using the ``-s``
+or ``--`` flags.  Either flag takes the name of the configuration variable
+followed by an equals character (``=``)' and the value to assign to it.
+This value must be a legal Chapel literal for the type of the variable.
+(Exception: for a string literal, the surrounding quotes are implicit.)
+In our current implementation, no extra spaces may appear between
+these elements.  Thus, the general form is:
+
+  .. code-block:: sh
+
+    ./a.out --<cfgVar>=<val>
+
+or:
+
+  .. code-block:: sh
+
+    ./a.out -s<cfgVar>=<val>
+
+As an example, compile the hello2-module.chpl example which prints a
+user-definable message:
+
+  .. code-block:: sh
+
+    chpl -o hello2-module $CHPL_HOME/examples/hello2-module.chpl
+
+This program defines a configuration constant, ``message``, indicating
+the string to print to the console, set to ``"Hello, world!"`` by
+default.  To override the default, you can use:
+
+  .. code-block:: sh
+
+    ./hello2-module --message='Hello, Chapel users!'
+
+or:
+
+  .. code-block:: sh
+
+    ./hello2-module -smessage='Hello, Chapel users!'
+
+The default value of a configuration constant or variable can also be
+overridden at compilation time with a ``-s`` option. (The surrounding
+quotes must be provided for a string literal.) For example:
+
+  .. code-block:: sh
+
+    cd $CHPL_HOME/examples
+    chpl -smessage='"Hello from the Chapel compiler"' hello2-module.chpl
+
+or:
+
+  .. code-block:: sh
+
+    chpl -s message="'Hello from the Chapel compiler'" hello2-module.chpl
+
+The compiler-established default can still be overridden when
+executing the program, as shown above.
+
+Chapel programs can also accept C-like command line arguments to their
+``main()`` procedure in addition to the aforementioned configuration
+variables. See :doc:`technotes/main` for more information.
+
+
+-----------------------------
+Setting the Number of Locales
+-----------------------------
+
+For multi-locale Chapel executions, the number of locales on which to
+execute a program is specified on the executable's command-line.  This
+can be set either using the -nl flag, or by assigning to the built-in
+numLocales configuration constant using the normal mechanisms.  So, to
+execute on four locales, one could use:
+
+  .. code-block:: sh
+
+    ./a.out -nl 4
+
+or:
+
+  .. code-block:: sh
+
+    ./a.out --numLocales=4
+
+or:
+
+  .. code-block:: sh
+
+    ./a.out -snumLocales=4
+
+For users running with ``$CHPL_COMM=none`` (the default), only one
+locale can be used.  See :doc:`multilocale` for more
```

Comment:
We are going to leave cross-references out until all READMEs are converted. This file won't actually exist in the sphinx project until it has been converted to reST.